modules/gmc-agent/src/main/java/org/gridgain/action/SessionRegistry.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gridgain.action;
+
+import org.apache.ignite.IgniteAuthenticationException;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.IgniteSystemProperties;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.gridgain.agent.AgentUtils;
+
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static org.apache.ignite.IgniteSystemProperties.IGNITE_REST_SECURITY_TOKEN_TIMEOUT;
+import static org.apache.ignite.IgniteSystemProperties.IGNITE_REST_SESSION_TIMEOUT;
+
+/**
+ * Session registry.
+ */
+public class SessionRegistry {
+    /** The default interval used to invalidate sessions, in seconds. */
+    private static final int DFLT_SES_TOKEN_INVALIDATE_INTERVAL = 5 * 60;
+
+    /** Default session timeout, in seconds. */
+    private static final int DFLT_SES_TIMEOUT = 30;
+
+    /** Instance. */
+    private static volatile SessionRegistry instance;
+
+    /** Context. */
+    private final GridKernalContext ctx;
+
+    /** Logger. */
+    private final IgniteLogger log;
+
+    /** SessionId-Session map. */
+    private final ConcurrentMap<UUID, Session> sesId2Ses = new ConcurrentHashMap<>();
+
+    /** Session time to live. */
+    private final long sesTtl;
+
+    /** Interval to invalidate session tokens. */
+    private final long sesTokTtl;
+
+    /**
+     * @param ctx Context.
+     */
+    private SessionRegistry(GridKernalContext ctx) {
+        this.ctx = ctx;
+        this.log = ctx.log(SessionRegistry.class);
+
+        sesTtl = IgniteSystemProperties.getLong(IGNITE_REST_SESSION_TIMEOUT, DFLT_SES_TIMEOUT) * 1000;
```

Comment:
I think we should not use options from REST.
And also all timeouts should be in milliseconds.

Reply:
I'll move this in ManagementConfiguration